lib/serializers/item.js

```diff
@@ -134,9 +134,12 @@ var fromMarcJson = (object, datasource) => {
     if (barcode) builder.add(fieldMapper.predicateFor('Identifier'), { id: barcode.value, type: 'bf:Barcode' }, 0, { path: barcode.path })
 
     // Callnum
-    var callnum = null
+    let callnum
     // First pull it from varfield:
-    if (object.varField('852', ['h'])) callnum = { value: object.varField('852', ['h'])[0], path: '852 $h' }
+    const shelfMarkMapping = fieldMapper.getMapping('Call number')
+    const shelfMarkMarc = shelfMarkMapping.paths[0].marc
+    const shelfMarkSubfields = shelfMarkMapping.paths[0].subfields
+    if (object.varField(shelfMarkMarc, ['h'])) callnum = { value: object.varField(shelfMarkMarc, shelfMarkSubfields)[0], path: shelfMarkSubfields.map((s) => `$${s}`).join(' ') }
```

Comment:
Is there a reason the conditional is `if (object.varField(shelfMarkMarc, ['h']))` instead of `if (object.varField(shelfMarkMarc, shelfMarkSubfields))`. The value of `shelfMarkSubfields` is currently `[h]`, so it's equivalent for now, but that may change.

Reply:
yeah, since the `h` subfield contains the actual call number, with other subfields (`k` and `i` mainly) containing prefix or suffix values, my thought was that without the `h` subfield this would represent an invalid MARC record and should be skipped. I could be wrong about that, with perhaps with being call numbers cataloged in non-standard ways? But it seems like we have/do assume that the `h` field really represents the shelfMark